### PR TITLE
Serviceaccounts: Add serviceaccount docs

### PR DESCRIPTION
**Pull request by eleijonmarck**
**What this PR does / why we need it**:
This introduces a page in the documentation for interacting with serviceaccounts.
- This adds a skeleton for where we will place the rest of the docuemntation
- Adds documentation for deletion of serviceaccounts

**Which issue(s) this PR fixes**:
https://github.com/grafana/grafana-enterprise/issues/2233

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/sources/manage-users/serviceaccount/serviceaccount.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
++++
+title = "What are service accounts"
+weight = 200
++++
+
+What are service accounts?
+
+A service account is a special kind of user used by an application or compute workload, which are users that are intended for programmatic use either internally for a Grafana feature or through an API key. Applications use service accounts to make authorized API calls, authorized as the service account itself.
+
+For example, a all reporting could be setup and run as a service account from your application, and that account can be given permissions to access the resources it needs. This way the service account is the identity of the service, and the service account's permissions control which resources the service can access.
```
Comment (osg-grafana) [inline docs/sources/manage-users/serviceaccount/serviceaccount.md:10]: "a all"?

File: docs/sources/manage-users/serviceaccount/serviceaccount.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+A service account is identified by its login name, which is unique to the entire sweet of organizations. The name is set upon creation of the service account.
```
Comment (osg-grafana) [inline docs/sources/manage-users/serviceaccount/serviceaccount.md:12]:
```suggestion
A service account is identified by its login name, which is unique to the entire suite of organizations. The name is set upon creation of the service account.
```

File: docs/sources/manage-users/serviceaccount/serviceaccount.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+Differences between a service account and a user account
+Service accounts differ from user accounts in a few key ways:
+
+- Service accounts do not have passwords, and cannot log in via browsers or cookies.
+- Service accounts are associated with private/public RSA key-pairs that are used for authentication to Grafana.
+- You can let other users or service accounts impersonate a service account.
+- Service accounts lives on a organizational level, but it does not restrict them from acting cross organizational, such as adding/editing user information.
```
Comment (osg-grafana) [inline docs/sources/manage-users/serviceaccount/serviceaccount.md:20]:
```suggestion
- Service accounts lives on a organizational level, but it does not restrict them from acting cross organizational, such as adding or editing user information.
```

**Review by osg-grafana — CHANGES_REQUESTED** [review]
sweet

File: docs/sources/http_api/serviceaccount.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
++++
+title = "Service account HTTP API"
+description = "Grafana Service account HTTP API"
+keywords = ["grafana", "http", "documentation", "api", "serviceaccount"]
```
Comment (osg-grafana) [inline docs/sources/http_api/serviceaccount.md:4]:
```suggestion
keywords = ["grafana", "http", "documentation", "api", "service account"]
```

Comment (osg-grafana) [reply docs/sources/http_api/serviceaccount.md:4]: Actually, I am not sure about change suggestion, given that the API path is `serviceaccount` and the documentation is `service account`. The former is harder to read.

File: docs/sources/http_api/serviceaccount.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+aliases = ["/docs/grafana/latest/http_api/serviceaccount/"]
++++
+
+# Service account API
+
+This API allows you to interact programmatically with the [Service accounts]({{< relref "../manage-users/serviceaccount/_index.md" >}}).
+
+> If you are running Grafana Enterprise and have [Fine-grained access control]({{< relref "../enterprise/access-control/_index.md" >}}) enabled, for some endpoints you would need to have relevant permissions.
+> Refer to specific resources to understand what permissions are required
```
Comment (osg-grafana) [inline docs/sources/http_api/serviceaccount.md:13]:
```suggestion
** > Note:** If you are using Grafana Enterprise and have [Fine-grained access control]({{< relref "../enterprise/access-control/_index.md" >}}) enabled, for some endpoints you need to have relevant permissions. Refer to specific resources to understand what permissions are required.
```

File: docs/sources/http_api/serviceaccount.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+## Delete a service account
+
+`DELETE /api/serviceaccounts/:serviceaccountId`
+
+#### Required permissions
+
+See note in the [introduction]({{< ref "#user-api" >}}) for an explanation.
```
Comment (osg-grafana) [inline docs/sources/http_api/serviceaccount.md:21]:
```suggestion
For details, see the [introduction]({{< ref "#user-api" >}}).
```

Comment (osg-grafana) [reply docs/sources/http_api/serviceaccount.md:21]: @achatterjee-grafana, do all references need to use `relref`? @GrafanaWriter and @chri2547, this question is something that we ought to include in our Markdown conventions at https://github.com/grafana/grafana/blob/main/contribute/style-guides/documentation-markdown-guide.md.

File: docs/sources/http_api/serviceaccount.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+| Action                 | Scope              |
+| ---------------------- | ------------------ |
+| serviceaccounts:delete | serviceaccounts:\* |
+
+Deletes the given service account if it exists.
+
+**Example Request**:
```
Comment (osg-grafana) [inline docs/sources/http_api/serviceaccount.md:29]:
```suggestion
**Example request**:
```

File: docs/sources/http_api/serviceaccount.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+```http
+DELETE /api/serviceaccounts/1 HTTP/1.1
+Accept: application/json
+Content-Type: application/json
+Authorization: Bearer eyJrIjoiT0tTcG1pUlY2RnVKZTFVaDFsNFZXdE9ZWmNrMkZYbk
+```
+
+**Example Response**:
```
Comment (osg-grafana) [inline docs/sources/http_api/serviceaccount.md:38]:
```suggestion
**Example response**:
```

File: docs/sources/http_api/serviceaccount.md (hunk 6 above)
Comment (achatterjee-grafana) [reply docs/sources/http_api/serviceaccount.md:21]: For internal content within the same repo, yes.

File: docs/sources/http_api/serviceaccount.md (hunk 4 above)
Comment (eleijonmarck) [reply docs/sources/http_api/serviceaccount.md:4]: its a good point. I am not sure here if we want to separate the service account. Sorry to always point to google cloud, but here they separate service and account(s) in their documentation.

Maybe we should too?
https://console.cloud.google.com/iam-admin/serviceaccounts

Comment (eleijonmarck) [reply docs/sources/http_api/serviceaccount.md:4]: they also note plural and not singluar

Comment (eleijonmarck) [reply docs/sources/http_api/serviceaccount.md:4]: these keywords btw, can they be separate with a space?

Comment (eleijonmarck) [reply docs/sources/http_api/serviceaccount.md]: thx for the input

**Comment by osg-grafana** [comment]
What is the milestone for this?

**Review by osg-grafana — APPROVED** [review]
Approving documentation, but there looks to be a non-doc related issue with this PR still.

**Comment by eleijonmarck** [comment]
> What is the milestone for this?

It is for 8.4

**Comment by eleijonmarck** [comment]
> Approving documentation, but there looks to be a non-doc related issue with this PR still.

So this is really a skeleton for the next steps of creating a full blown documentation fro serviceaccounts

**Pull request merged**